src/onepay-checkout.js

```diff
@@ -666,12 +666,20 @@ function getHttpRequestInstance() {
   return new XMLHttpRequest();
 }
 
-function prepareOnepayHttpRequestParams() {
+function prepareOnepayHttpRequestParams(params) {
   let paramsUrl = 'channel=WEB';
-
-  if (typeof SmartPhone !== 'undefined' && (SmartPhone.isAndroid() || SmartPhone.isIOS())) {
+  if (typeof Smartphone !== 'undefined' && (Smartphone.isAndroid() || Smartphone.isIOS())) {
     paramsUrl = 'channel=MOBILE';
   }
+
+  if (params && Array.isArray(params)) {
+    params.forEach((param) => {
+      if (param && param.name && param.value) {
+        paramsUrl += '&' + param.name + '=' + param.value;
```

Comment:
This looks wrong. You need to escape the names and values using escapeURIComponent (otherwise any weird character will mess your url)

Reply:
By the way, using map() and join(‘&’) this code can be a bit cleaner. I’m on mobile and can’t write proper code here, but you can ask @eugeniolopezramos on how to do it if you don’t know what I’m talking about :)